apps/dashboard/app/views/layouts/nav/_featured_apps.erb

```diff
@@ -0,0 +1,38 @@
+<li class="dropdown" title="<%= group.title %>" >
+  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false"><%= group.title %><span class="caret"></span></a>
+  <ul class="dropdown-menu">
+    <% OodAppGroup.groups_for(apps: group.apps, group_by: :subcategory).each_with_index do |g, index| %>
+      <%- rendering_pinned_apps = g.apps.first.subcategory == "Pinned Apps" -%>
+      <%- if rendering_pinned_apps && g.apps.size > Configuration.pinned_apps_menu_length -%>
+        <%- title = "#{g.title} (showing #{Configuration.pinned_apps_menu_length} of #{g.apps.size})" %>
+      <%- else -%>
+        <%- title = g.title.to_s -%>
+      <%- end -%>
+
+      <%= content_tag "li", "", class: "divider", role: "separator" if index > 0 %>
+      <%= content_tag "li", title, class: "dropdown-header" unless g.title.empty? %>
+      <% g.apps.each_with_index do |app, app_index| %>
```

Comment:
instead of each_with_index and a break if `app_index >= Configuration.pinned_apps_menu_length` use take https://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-take

```ruby
<% g.apps.take(Configuration.pinned_apps_menu_length) do |app| %>
```

Reply:
Done.